src/Relation/Binary/TypeClasses/Examples.agda

```diff
@@ -0,0 +1,47 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Usage examples of typeclasses for binary relations
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Relation.Binary.TypeClasses.Examples where
+
+open import Relation.Nullary
+open import Relation.Binary.PropositionalEquality
+open import Relation.Binary.TypeClasses
+
+open import Data.Bool hiding (_≟_)
+open import Data.Bool.Instances
+open import Data.List
+open import Data.List.Instances
+open import Data.Nat hiding (_≟_)
+open import Data.Nat.Instances
+open import Data.Product
+open import Data.Product.Instances
+open import Data.Unit hiding (_≟_)
+open import Data.Unit.Instances
+open import Data.Vec
+open import Data.Vec.Instances
+
+test-Dec≡-Bool : Dec (true ≡ true)
+test-Dec≡-Bool = true ≟ true
+
+test-Dec≡-Nat : Dec (0 ≡ 1)
+test-Dec≡-Nat = 0 ≟ 1
+
+test-Dec≡-List : Dec (_≡_ {A = List ℕ} (1 ∷ 2 ∷ []) (1 ∷ 2 ∷ []))
```

Comment:
Probably more readable to write this as `Dec {A = List ℕ} (1 ∷ 2 ∷ [] ≡ 1 ∷ 2 ∷ [])`. Same for the ones below

Reply:
If only `Dec` would take an argument `{A = ...}` then I could do that.

Reply:
Haha doh, yes sorry! Pattern recognition failure there. I spend a lot of time fixing `Decidable (_≡_ {A = List A})` patterns in PRs.